client/src/PlayerCommands.cpp

```diff
@@ -44,31 +44,31 @@ bool PlayerCommands::isMovingRight() const {
 PlayerCommands::PlayerCommands() : movingUp(false), movingDown(false), movingLeft(false), movingRight(false),
                                    quadTreeDebug(false) {}
 
-Player *PlayerCommands::getPlayer() const {
+const std::weak_ptr<Player> &PlayerCommands::getPlayer() const {
     return player;
 }
 
-void PlayerCommands::setPlayer(Player *player) {
+void PlayerCommands::setPlayer(const std::weak_ptr<Player> &player) {
     PlayerCommands::player = player;
 }
 
 void PlayerCommands::updatePlayer(sf::Time deltaTime) {
     sf::Vector2f movement(0.f, 0.f);
     if (movingUp)
-        movement.y -= player->getSpeed();
+        movement.y -= player.lock()->getSpeed();
     if (movingDown)
-        movement.y += player->getSpeed();
+        movement.y += player.lock()->getSpeed();
```

Comment:
Exactly, your check will be done once and then you have the guarantee that your resource won't be freed until your shared_ptr exist.

That's the magic of smart pointers :)

But you should also avoid storing these shared_ptr for a long time (aka : a time superior to a function execution)